src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -124,7 +121,12 @@ class LazyVals extends MiniPhaseTransform with SymTransformer {
 
 
   override def transformStats(trees: List[tpd.Tree])(implicit ctx: Context, info: TransformerInfo): List[tpd.Tree] = {
-    val (holders, stats) = trees.partition { _.symbol.flags == containerFlags}
+    val (holders, stats) =
+      atGroupEnd { implicit ctx: Context =>
+        trees.partition {
+          _.symbol.flags == containerFlags
```

Comment:
> Because any read of these fields would cause the Touched flag to be set. 

Would it? The only place where I can see `Touched` flags is set is `SymDenotations.completeFrom`. As none of LazyVals defined symbols has a `LazyType` this method will not be called. I've just made an assetion that there are no statements that have `containerFlags | Touched` flags set, it never triggered for whole dotty test suite.